### PR TITLE
fix bug #2485

### PR DESCRIPTION
**Pull request by yaohcn**
fix bug #2485

### CODE REVIEW

**Comment by mishmosh** [comment]
Thanks @yaohcn. Assigning to @ZenGround0 for first review.

File: core/message_pool.go [hunk 1]
```diff
@@ -181,10 +181,13 @@ func (pool *MessagePool) timeoutMessages(ctx context.Context, store chain.BlockP
...
 	}
 
+	pool.lk.Lock()
+	defer pool.lk.Unlock()
+
 	// remove all messages added before minimumHeight
 	for cid, msg := range pool.pending {
 		if msg.addedAt < minimumHeight {
-			pool.Remove(cid)
+			delete(pool.pending, cid)
```
Comment (anorth) [inline core/message_pool.go]: This misses the metric update in `Remove()`.

It is unfortunate that `timeoutMessages` uses the pool's private API, which makes errors like this all too easy (not your fault). I think a better fix here is to use the exported `Pending()` at line 188 then stick with `Remove` here (and no need for the extra lock).

**Comment by anorth** [comment]
I think this may be accidentally(?) fixed by #2522 too

File: core/message_pool.go (hunk 1 above)
Comment (yaohcn) [reply core/message_pool.go]: Pending() return only field: message *types.SignedMessage. but here need the field: addedAt uint64.

**Comment by yaohcn** [comment]
> I think this may be accidentally(?) fixed by #2522 too

yep, closed.

**Pull request closed without merging**